File: exercism_data/src/456.rb
class Phrase

  def initialize string
    @string = string
  end

  def word_count
    words = Hash.new(0)
    @string.split(/\W*\s\W*|,|\.|\W{2,}/).each { |word| words[word.downcase] += 1 }
    words
  end
end

File: assignments/python/exercism_wc/src/456.py
def word_count(input_list):
    my_dict = {} # create an empty dictionary
    for word in input_list.split(): # split and operate on each item
        if word not in my_dict: # for new words
            my_dict[word] = 1 # create a value of 1
        else:
            my_dict[word] += 1 # for existing words, increment the value 
    return my_dict # return the dictionary
